python/prophet/numpyro_model.py

```diff
@@ -0,0 +1,138 @@
+from jax import jit
+from jax.lax import scan
+import jax.numpy as jnp
+from jax.scipy.special import expit
+import numpy as np
+import numpyro.distributions as dist
+from numpyro.infer import MCMC, NUTS
+from numpyro import sample, plate, deterministic
+from numpyro.infer import SVI
+from numpyro.infer.autoguide import AutoDelta, Trace_ELBO
+from numpyro.infer.initialization import init_to_value
+from numpyro.optim import Minimize
+from jax import random
+
+
+def construct_changepoint_matrix(t, t_change, T, S):
+    A = np.zeros((T, S))
+    a_row = np.zeros(S)
```

Comment:
Exactly. You can't modify a jax array in-place. Since this is only called once, I figured it was easier and less risky to convert the stan code to python with regular numpy and then convert to jax at the end.